aws_compliance.py

```diff
@@ -91,6 +98,114 @@ def should_skip_bucket(bucket):
+        else:
+            offenders.append(str(VULS_REPORT_BUCKET) + ":Error listing objects")
+            failReason = "Error listing objects: " + str(e)
+    if response['Contents']:
+            for object in response['Contents']:
+                  if object['Key'].split('.')[-1] == "json":
+                      report = S3_CLIENT.get_object(Bucket=VULS_REPORT_BUCKET,Key=object['Key'])
+                      report_body = json.loads(report['Body'].read())
+                      known_cves = report_body.get('KnownCves')
+                      unknown_cves = report_body.get('UnknownCves')
+                      if report_body.get('Optional'):
+                          options = dict(report_body['Optional'])
+                          if options.get('environment'):
+                              env = options.get('environment')
+                              if env not in offenders:
+                                  offenders.append(env)
+                      if known_cves:
+                          for known_cve in known_cves:
+                              gen_cve_summary(cve_summary,known_cve,report_body['ServerName'])
+                      if unknown_cves and VULS_IGNORE_UNSCORED_CVE is False:
+                          for unknown_cve in unknown_cves:
+                              gen_cve_summary(cve_summary,unknown_cve,report_body['ServerName'])
+            final = final_cve_summary(cve_summary)
+            if len(final) >= 0:
+                result = False
+                failReason = final
+    else:
+        result = False
+        failReason = "No Vuls reports found for today in " + VULS_REPORT_BUCKET
+
+    return {'Result': result, 'failReason': failReason, 'Offenders': offenders, 'ScoredControl': scored, 'Description': description, 'ControlId': control}
+
+def severity_to_num(severity):
+    if severity == 'unknown':
+        return 1
+    elif severity == 'low':
+        return 2
+    elif severity == 'medium':
+        return 3
+    elif severity == 'high':
+        return 4
+
+
+def final_cve_summary(cve_summary):
+    for k,v in cve_summary.items():
+        if severity_to_num(v['severity']) < severity_to_num(VULS_MIN_ALERT_SEVERITY):
+           del cve_summary[k]
+    return cve_summary
+
+
+def gen_cve_summary(summary,cve,servername):
+   cve_id = cve['CveDetail']['CveID']
+   summary[cve_id] = summary.get(cve_id) or dict()
+   if not summary[cve_id].get('score'):
+       summary[cve_id]['score'] = cve_score(cve)
+   if not summary[cve_id].get('severity'):
+       summary[cve_id]['severity'] = cve_severity(cve)
+   summary[cve_id]['instances'] = summary[cve_id].get('instances') or []
+   summary[cve_id]['instances'].append(servername)
+
+
+def cve_score(cve):
+    if cve['CveDetail']['Nvd']['Score'] > 0:
+        return cve['CveDetail']['Nvd']['Score']
+    elif cve['CveDetail']['Jvn']['Score'] > 0:
+        return cve['CveDetail']['Nvd']['Score']
```

Comment:
The condition is looking at the JVN score, but setting NVD in the return value. It looks wrong; if it isn't, please add a comment to the code as to why this is this way for future generations.